tests/gateways/test_connection.py

```diff
@@ -272,9 +271,13 @@ def test_get_session_with_url_pattern(mocker, channel_settings_url, expect_match
         "conda.gateways.connection.session.get_channel_name_from_url",
         return_value=channel_url,
     )
-    mock_context: Context = mocker.patch("conda.gateways.connection.session.context")
-    mock_context.channel_settings = (
-        {"channel": channel_settings_url, "auth": "dummy_one"},
+    mocker.patch(
+        "conda.base.context.Context.channel_settings",
+        new_callable=mocker.PropertyMock,
+        return_value=[{"channel": channel_settings_url, "auth": "dummy_one"}],
+    )
+    get_auth_handler = mocker.patch(
+        "conda.plugins.manager.CondaPluginManager.get_auth_handler"
```

Comment:
This is a targeted mocking for a specific value within the context object. Since it is so targeted we also need to explicitly patch `get_auth_handler` as well for the test to do the same as before.